mocha-casperjs.js

```diff
@@ -1,15 +1,37 @@
 module.exports = function (Mocha, casper, utils) {
   var currentDone,
       currentTest,
+      currentError,
       f = utils.format,
 
   failTest = function(error) {
     casper.unwait()
     clearInterval(casper.checker)
-    casper.checker = null
-    if (currentDone && (!currentTest || !currentTest.state)) {
-      currentDone(error)
+
+    // the first error takes priority
+    if ( ! currentError ) {
+        currentError = error;
+    }
+    if ( currentTest.errors === undefined ) {
+        currentTest.errors = [error];
+    } else {
+        currentTest.errors.push(error);
     }
+    function reportError() {
+        casper.checker = null
+        if (currentDone && (!currentTest || !currentTest.state)) {
+            currentDone(currentError)
+        }
+    }
+
+    if ( casper.step < casper.steps.length ) {
+        casper.run(function() {
```

Comment:
IIRC casper doesn't work well calling `casper.run` multiple times. rather the reporting of the error should be moved down to the main `done()` call on (current) line 100.